actionpack/lib/action_dispatch/middleware/show_exceptions.rb

```diff
@@ -80,6 +81,15 @@ def fallback_to_html_format_if_invalid_mime_type(request)
         end
       end
 
+      def fallback_to_html_when_invalid_multipart(request, wrapper)
+        rack_error = wrapper.wrapped_causes.find { |wrapped| wrapped.exception.is_a?(::EOFError) }
+        return if rack_error.nil? || !request.form_data?
+
+        request.set_header "CONTENT_TYPE", "application/octet-stream"
+        # Required for Rack 3
+        request.delete_header(Rack::RACK_REQUEST_FORM_ERROR) if defined?(Rack::RACK_REQUEST_FORM_ERROR)
```

Comment:
https://github.com/rack/rack/blob/d73095f6c03007ceb7ccb4f76ebb785899c0dcde/lib/rack/request.rb#L502-L503

Rack 3 re-raise if this header is set.